### DISCUSSION TITLE
Twill install failes to complete migrations

### DISCUSSION THREAD

**Discussion by radovanobal** [area17/twill#855] [Q&A]
Running the command `php artisan twill:install` results in an error:

-----------------------------------------------------------------------------------------------
Migrating: 2020_02_09_000007_create_twill_default_activity_log_table

   Illuminate\Database\QueryException 

  SQLSTATE[42000]: Syntax error or access violation: 1103 Incorrect table name '' (SQL: create table `` (`id` int unsigned not null auto_increment primary key, `log_name` varchar(255) null, `description` text not null, `subject_id` int null, `subject_type` varchar(255) null, `causer_id` int null, `causer_type` varchar(255) null, `properties` text null, `created_at` timestamp null, `updated_at` timestamp null) default character set utf8mb4 collate 'utf8mb4_unicode_ci')

  at vendor/laravel/framework/src/Illuminate/Database/Connection.php:671
    667|         // If an exception occurs when attempting to run a query, we'll format the error
    668|         // message to include the bindings with SQL, which will make this exception a
    669|         // lot more helpful to the developer instead of just the database's errors.
    670|         catch (Exception $e) {
  > 671|             throw new QueryException(
    672|                 $query, $this->prepareBindings($bindings), $e
    673|             );
    674|         }
    675| 

      +46 vendor frames 
  47  artisan:37
      Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
-----------------------------------------------------------------------------------------------

Running in docker environment:

mysql: 5.7
php: 7.4.9
Laravel: 7.27.0

**Comment by ersin-demirtas** [comment]
Check your migration file, you have deleted or comment out the table name.

**Comment by radovanobal** [comment]
I have not touched any of twill files. This results from a fresh composer install followed by twill:install.

**Comment by antonioribeiro** [comment]
The migrator get the table name from config:

```
config('activitylog.table_name')
```

So I believe if you do `php artisan tinker` and run this same command, it should return null or empty. This may be a start for you to debug it.

**Comment by radovanobal** [comment]
I am sorry but if there is additional steps required after a fresh install of laravel and then a fresh install of latest twill then the documentation is lacking.

**Comment by sambaynham** [comment]
I just encountered this issue. In order to get it working I had to A: Publish spatie/activity_log's service provider, B:Clear my config cache and C: Enter the vendor directory, comment out Twill's version of the activity_log migration and then run php artisan migrate. After that, I uncommented Twill's activity log migration, as it already had a check to see that the table wasn't populated. After that, the install ran. What puzzles me is why Twill is duplicating the migration of a dependency package?

**Comment by ptrckvzn** [comment]
Hi @radovanobal, could you provide steps to reproduce the issue? Thanks
